projects/plugins/crm/includes/ZeroBSCRM.GeneralFuncs.php

```diff
@@ -527,6 +527,15 @@ function zeroBSCRM_findAB($html,$first,$nextStr,$fallbackCloser='</'){
  */
 function zeroBSCRM_retrieveFile( $url, $filepath, $args = array() ) {
 
+	$large_font_files = array( 'NotoSansSimplifiedChinese.zip', 'NotoSerifDisplay.zip', 'NotoSans.zip', 'NotoSansDisplay.zip', 'NotoSansTaiwan.zip', 'NotoSansHongKong.zip', 'NotoSansJapanese.zip', 'NooSansKorean.zip', 'NotoSansMono.zip' );
+	// Several large font files may timeout when downloading. Increase the timeout for these files.
+	foreach ( $large_font_files as $file ) {
+		if ( str_ends_with( $url, $file ) ) {
+
+			add_filter( 'http_request_timeout', 'custom_large_font_timeout' );
+		}
+	}
+
```

Comment:
This works. Did you consider moving this into `retrieve_and_install_specific_font()`, though? I think it'd work better here:

https://github.com/Automattic/jetpack/blob/2a2fb2513a591ca5ebb0b5ce239c81b4cf0cae1b/projects/plugins/crm/includes/jpcrm-fonts.php#L479

This would mean you no longer need `str_ends_with()` (or even a loop), as you have the actual filename already.

Reply:
Did not consider that, I made an assumption the timeout filter needed to be in the same function but I can see it doesn't! Thank you.